evaluation.py

```diff
@@ -14,84 +14,82 @@
-    player_pairs = [(b, w) for b, w in zip(black_players, white_players)]
+    black = MCTSPlayer(
+        black_net, verbosity=verbosity, two_player_mode=True, num_parallel=SIMULTANEOUS_LEAVES)
+    white = MCTSPlayer(
+        white_net, verbosity=verbosity, two_player_mode=True, num_parallel=SIMULTANEOUS_LEAVES)
 
-    done_pairs = []
+    black_name = os.path.basename(black_net.save_file)
+    white_name = os.path.basename(white_net.save_file)
 
-    # The number of moves that have been played
-    num_moves = 0
+    for i in range(games):
+        num_move = 0  # The move number of the current game
 
-    for black, white in player_pairs:
         black.initialize_game()
         white.initialize_game()
 
-    # The heart of the game-playing loop. Each iteration through the while loop
-    # plays one move for each player. That means we:
-    #   - Do a bunch of MTCS readouts (for each active player, for each game)
-    #   - Play a move (for each active player, for each game)
-    #   - Remove any finished player-pairs
-    while player_pairs:
-        start = time.time()
-
-        for _ in range(readouts):
-            leaves = [pair[num_moves % 2].root.select_leaf()
-                      for pair in player_pairs]
-            probs, vals = (black_net, white_net)[num_moves % 2].run_many(
-                [leaf.position for leaf in leaves])
-
-            for pair, leaf, prob, val in zip(player_pairs, leaves, probs, vals):
-                leaf.incorporate_results(
-                    prob, val, up_to=pair[num_moves % 2].root)
-
-        # print some stats on the search
-        if verbosity >= 3:
-            print(player_pairs[0][0].root.position)
-
-        for black, white in player_pairs:
-            active = white if num_moves % 2 else black
-            inactive = black if num_moves % 2 else white
+        while True:
+            start = time.time()
+            for _ in range(readouts):
+                player = white if num_move % 2 else black
+                player.tree_search()
+
+            # print some stats on the search
+            if verbosity >= 3:
+                print(player.root.position)
+
+            active = white if num_move % 2 else black
+            inactive = black if num_move % 2 else white
             # First, check the roots for hopeless games.
             if active.should_resign():  # Force resign
-                continue
+                active.set_result(-1 *
+                                  active.root.position.to_play, was_resign=True)
+                inactive.set_result(
+                    active.root.position.to_play, was_resign=True)
+
+            if active.is_done():
+                fname = "{:d}-{:s}-vs-{:s}-{:d}.sgf".format(int(time.time()),
+                                                            white_name, black_name, i)
+                with open(os.path.join(sgf_dir, fname), 'w') as _file:
+                    sgfstr = sgf_wrapper.make_sgf(active.position.recent,
```

Comment:
I could have sworn I had some cute code snippet that would splice in alternate comments from both players so you'd get to see both players' debug perspectives. Let me figure out where I put it...